extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ClientContextResolverHandler.java

```diff
@@ -0,0 +1,180 @@
+package io.quarkus.rest.client.reactive.deployment;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.LinkedHashMap;
+
+import jakarta.ws.rs.Priorities;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+import org.jboss.resteasy.reactive.client.impl.RestClientRequestContext;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.gizmo.SignatureBuilder;
+import io.quarkus.rest.client.reactive.runtime.ResteasyReactiveContextResolver;
+import io.quarkus.runtime.util.HashUtil;
+
+/**
+ * Generates an implementation of {@link ResteasyReactiveContextResolver}
+ * from an instance of {@link jakarta.ws.rs.ext.Provider}
+ */
+class ClientContextResolverHandler {
```

Comment:
I would like some more general comments about this and why it's done like this. 
Asking because in a month, I am pretty sure no one will remember why we did it this way

Reply:
added

Reply:
Are you sure you have pushed the change?

Reply:
Now yes